packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -40,15 +33,131 @@ export type CustomSelectProps = {
 	/**
 	 * The size of the control.
 	 *
-	 * @default 'default'
+	 * @default default
 	 */
-	size?: 'default' | 'small';
+	size?: 'compact' | 'default' | 'small';
 	/**
 	 * Can be used to externally control the value of the control.
 	 */
 	value?: string | string[];
 };
 
+export type _CustomSelectProps = {
+	/**
+	 * The child elements. This should be composed of CustomSelectItem components.
+	 */
+	children: React.ReactNode;
+	/**
+	 * Used to visually hide the label. It will always be visible to screen readers.
+	 *
+	 * @default false
+	 */
+	hideLabelFromVision?: boolean;
+	/**
+	 * Label for the control.
+	 */
+	label: string;
+};
+
+export type CustomSelectProps = _CustomSelectProps &
+	Exclude< CustomSelectButtonProps, { size: 'small' } >;
```

Comment:
`CustomSelectButtonProps` itself is not a union type so I don't think we can exclude it directly. How about something like this:

```suggestion
export type CustomSelectProps = _CustomSelectProps &
	Omit< CustomSelectButtonProps, 'size' > & {
		/**
		 * The size of the control.
		 *
		 * @default 'default'
		 */
		size?: Exclude< CustomSelectButtonProps[ 'size' ], 'small' >;
	};
```